### DISCUSSION TITLE
Suggested new names for the "Batch"

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#228] [General]
As discussed in https://github.com/sablier-labs/v2-periphery/discussions/225, in future releases of the protocol, this contract will do more than just batch-create streams. Accordingly, we need to find a new name for it. Suggestions:

1. `Router`
2. `Actions`
3. `Wrapper`

**Comment by PaulRBerg** [comment]
Any thoughts on this, @sablier-labs/solidity?

**Reply by andreivladbrg** [reply]
I can’t find exactly where we discussed this, but I remember considering adding a `Batch` contract similar to the one in [Flow](https://github.com/sablier-labs/flow/blob/7e2853737724173ebcb408f691bb08009a71631c/src/abstracts/Batch.sol) to the `Lockup` contracts as well. This would allow us to remove the current version.

**Reply by PaulRBerg** [reply]
Good idea

**Comment by smol-ninja** [comment]
Why not consider renaming it when it does more than just batch-create streams?

**Reply by PaulRBerg** [reply]
Yeah fair enough. That, or do what Andrei suggested above.

**Comment by PaulRBerg** [comment]
Closing as we have moved this to Lockup and `Batch` is a good name.
